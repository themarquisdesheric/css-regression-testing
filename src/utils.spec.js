import getComputedPropertiesToString, { getComputedProperties } from './utils';

const mockGetComputedStyle = window.getComputedStyle = jest.fn();
const styleDeclaration = {
  'background-position': '0% 0%',
  'background-repeat': 'repeat',
  'background-size': 'auto',
  '0': 'background-position',
  '1': 'background-repeat',
  '2': 'background-size',
  'length': 3
};

const mockStyleDeclaration = {
  ...styleDeclaration,
  getPropertyValue: (cssProperty) =>
    styleDeclaration[cssProperty]
};

mockGetComputedStyle.mockReturnValue(mockStyleDeclaration);

describe('utilities', () => {
  test('getComputedPropertiesToString should return a string of the computed properties', () => {
    const computedProperties = getComputedPropertiesToString('some node', null);

    expect(computedProperties).toMatchSnapshot();
  });

  test('getComputedProperties should return a string of the computed properties', () => {
    const computedProperties = getComputedProperties('some node', null);

    expect(computedProperties).toMatchSnapshot();
  });
});
