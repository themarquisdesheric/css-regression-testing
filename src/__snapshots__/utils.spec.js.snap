// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utilities getComputedProperties should return a string of the computed properties 1`] = `
Object {
  "background-position": "0% 0%",
  "background-repeat": "repeat",
  "background-size": "auto",
}
`;

exports[`utilities getComputedPropertiesToString should return a string of the computed properties 1`] = `
"background-position: 0% 0%;
background-repeat: repeat;
background-size: auto;
"
`;
